File: cpp/14.longest_common_prefix.cpp
#include <string>
#include <vector>
#include <algorithm>
class Solution {
public:
    std::string longestCommonPrefix(std::vector<std::string>& strs) {
        if (strs.size() == 1)
            return strs[0];

        std::sort(strs.begin(), strs.end());
        std::string ret = "", first = strs.front(), last = strs.back();
        for (int i = 0; i < first.length(); i++) {
            if (first[i] == last[i])
                ret += first[i];
            else break;
        }
        return ret;
    }
};

File: py/14.longest_common_prefix.py
from typing import List
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) < 0:
            return strs[0]
        ans = ""
        strs.sort()
        for i, value in enumerate(strs[0]):
            if value == strs[len(strs) - 1][i]:
                ans += value
            else:
                break
        return ans
